### ISSUE TITLE
How to record most used hubot commands or scripts or regex patterns ?

### ISSUE THREAD

**Issue by ravikiranj** [github/hubot#1079]
Hi,
    I'm trying to find out what are the most used hubot commands or scripts. Is there any way I can dump some stats like below ?

| hubot command  | count |
| ------------- | ------------- |
| hubot image  | 100  |
| hubot animate | 90  |
| hubot ship it  | 1000  |
| hubot kitten me  | 2000  |

Can you point me how I can hook onto the regex patterns that a particular hubot instance responds to ?

**Comment by cycomachead** [comment]
My best suggestion would be that you could use the new middleware API, and just store an object in the brain that's a regex : count map. When using the middleware API, each listener is supposed to have an options object with an `id` key. If that exists, that'd be easy to access but since many things aren't updated, you'll still need a way of accessing the regex, which I'm sure is possible but I don't know how to do it off hand.

Check out these docs: https://github.com/github/hubot/blob/master/docs/scripting.md#middleware
Also, look at the hubot-rate-limit example project linked.

**Comment by michaelansel** [comment]
I just built a metrics collector based on [this comment](https://github.com/github/hubot/pull/803#issuecomment-64054374). I might get around to open sourcing it, but it's also pretty trivial to build your own (to integrate with whatever metrics system you have).

The biggest shortcoming of my method is that it depends on having listener IDs, which are still relatively rare. I ended up just adding IDs to the internal scripts I care about most.
